packages/back-end/src/entities/database.ts

```diff
@@ -0,0 +1,234 @@
+import neo4j, { Driver } from "neo4j-driver";
+import { Person, PersonProperties } from "./person";
+
+export interface DatabaseService {
+  findPersonById(id: number): Promise<Person>;
+  findPersonByName(name: string): Promise<Person>;
+
+  findFather(id: number): Promise<Person>;
+  findFather(person: Person): Promise<Person>;
+
+  findMother(id: number): Promise<Person>;
+  findMother(person: Person): Promise<Person>;
+
+  findPartners(id: number): Promise<Person[]>;
+  findPartners(person: Person): Promise<Person[]>;
+
+  findChildren(id: number): Promise<Person[]>;
+  findChildren(person: Person): Promise<Person[]>;
+
+  updatePerson(person: PersonProperties): Promise<void>;
+  createPerson(person: PersonProperties): Promise<void>;
+  setFather(person: Person, father: Person): Promise<void>;
+  setMother(person: Person, mother: Person): Promise<void>;
+  addPartner(person: Person, partner: Person): Promise<void>;
+  removePartner(person: Person, partner: Person): Promise<void>;
+}
+
+export default class DatabaseServiceImpl implements DatabaseService {
+  private driver: Driver;
+
+  constructor(driver: Driver) {
+    this.driver = driver;
+  }
+
+  async findPersonById(id: number): Promise<Person> {
+    const session = this.driver.session();
+
+    try {
+      const result = await session.run(
+        "MATCH (p:Person) WHERE p.id = $id RETURN p",
+        { id }
+      );
+
+      const person = result.records[0].get("p");
+
+      return person;
+    } finally {
```

Comment:
Is the finally still accessible if there is a return in the try block ?

Reply:
Yep